packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,815 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { benchmark, BenchmarkType } from "@fluid-tools/benchmark";
+import { unreachableCase } from "@fluidframework/common-utils";
+import {
+	createField,
+	EditableField,
+	FieldKinds,
+	getField,
+	isEditableField,
+	isUnwrappedNode,
+	namedTreeSchema,
+	singleTextCursor,
+	UnwrappedEditableField,
+	valueSymbol,
+} from "../../feature-libraries";
+import { brand } from "../../util";
+import { ITestTreeProvider, TestTreeProvider } from "../utils";
+import { ISharedTree } from "../../shared-tree";
+import {
+	FieldKindIdentifier,
+	fieldSchema,
+	GlobalFieldKey,
+	IForestSubscription,
+	JsonableTree,
+	LocalFieldKey,
+	moveToDetachedField,
+	NamedTreeSchema,
+	rootFieldKey,
+	rootFieldKeySymbol,
+	SchemaData,
+	TreeSchemaIdentifier,
+	UpPath,
+	ValueSchema,
+} from "../../core";
+
+/**
+ * Shapes of trees to test with.
+ */
+enum TreeShape {
+	/**
+	 * Each node is inserted directly 1 layer below the root node, making a wide tree.
+	 * Used to measure the performance of editing long sequences.
+	 */
+	Wide = 0,
+	/**
+	 * Each node is inserted one layer below the previous node, making a stick-like tree.
+	 * Used to measure the performance costs related to the length of paths.
+	 */
+	Deep = 1,
+}
+
+async function createSharedTrees(
+	schemaData: SchemaData,
+	data: JsonableTree[],
+	numberOfTrees = 1,
+): Promise<readonly [ITestTreeProvider, readonly ISharedTree[]]> {
+	const provider = await TestTreeProvider.create(numberOfTrees);
+	for (const tree of provider.trees) {
+		assert(tree.isAttached());
+	}
+	provider.trees[0].storedSchema.update(schemaData);
+	provider.trees[0].context.root.insertNodes(0, data.map(singleTextCursor));
+	await provider.ensureSynchronized();
+	return [provider, provider.trees];
+}
+
+const globalFieldKey: GlobalFieldKey = brand("foo");
+const localFieldKey: LocalFieldKey = brand("foo");
+const rootSchemaName: TreeSchemaIdentifier = brand("Test");
+
+function getTestSchema(fieldKind: { identifier: FieldKindIdentifier }): SchemaData {
```

Comment:
I think its clearer to name functions which create a new object based on the input starting with "create" (like the one above) rather than "get". "get" implies the result already exists, and the function finds and returns it (like in getLeafFromJSObject).

Reply:
removed `getTestSchema` function